planner/core/exhaust_physical_plans.go

```diff
@@ -448,9 +448,17 @@ func (p *LogicalJoin) constructIndexMergeJoin(
 			keyOffMap[idxOff] = i
```

Comment:
It seems that we can check `if i != idxOff` here?

Reply:
No, we can't. `keyOffMapList` has been sorted below.